### ISSUE TITLE
Typescript style-only imports are not resolving correctly

### ISSUE THREAD

**Issue by mjbvz** [Microsoft/TypeScript#25358]
_From @ksnyde on July 1, 2018 17:43_

- VSCode Version: Version 1.24.1 (1.24.1)
- OS Version: macOS 10.13.5

Steps to Reproduce:

1. use `tsc` to transpile (using TS 3.0.0-dev.20180603) project with dynamic import like:

    ```typescript
    private static _defaultDb: import("abstracted-firebase").RealTimeDB;
    ```

2. run `tsc` with the following `tsconfig.json`:

    ```typescript
    {
      "compilerOptions": {
        "declaration": true,
        "module": "esnext",
        "target": "es2017",
        "lib": ["es2017", "esnext.asynciterable", "es2015.reflect"],
        "moduleResolution": "node",
        "sourceMap": true,
        "noImplicitAny": true,
        "experimentalDecorators": true,
        "emitDecoratorMetadata": true,
        "outDir": "./dist/esnext",
        "removeComments": false,
        "typeRoots": ["./node_modules/@types"]
      },
      "include": ["src/**/*"],
      "exclude": ["node_modules/**/*", "node_modules/@firebase/**/*.d.ts"]
    }
    ```

No errors are reported but if you go into the `.d.ts` file where the import existed you'll find a transpilation similar to:

```typescript
    protected _db: import("../../../../../../Users/ken/mine/forest-fire/firemodel/src/decorators/schema").RealTimeDB;
```

Does this issue occur when all extensions are disabled?: Yes

_Copied from original issue: Microsoft/vscode#53400_

**Comment by DanielRosenwasser** [comment]
Looks like one of

* #24599 
* #25345
* #25186

Can you try out our nightly (`npm install -g typescript@next`) and see if you're able to reproduce the issue?

**Comment by andy-ms** [comment]
Fix should just have been merged with #25364, please try with `typescript@next`.
